docker.go

```diff
@@ -189,7 +182,61 @@ func sanitize(s string) string {
 	return strings.Replace(s, "/", "-", -1)
 }
 
-func (i *dockerImage) makeRequest(method, url string, auth bool, headers map[string]string) (*http.Response, error) {
+type dockerImageSource struct {
+	ref             reference.Named
+	tag             string
+	registry        string
+	username        string
+	password        string
+	WWWAuthenticate string // Obtained by s.ping(); FIXME: why in GetManifest?
```

Comment:
will fix this, though, this whole interaction should go to vbatts/docker-utils one day